### PR TITLE
sci-libs/opencascade: add 7.7.1, 7.7.2, 7.8.0, 9999, drop obsolete blocks, add myself as maintainer

### PR DESCRIPTION
**Pull request by negril**
Closes: https://bugs.gentoo.org/923350
Closes: https://bugs.gentoo.org/911578
Closes: https://bugs.gentoo.org/908889

### CODE REVIEW

**Comment by negril** [comment]
Waiting for https://bugs.gentoo.org/924721

File: sci-libs/opencascade/opencascade-7.7.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,352 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# USE_{DRACO,FFMPEG,FREEIMAGE,FREETYPE,GLES2,OPENGL,OPENVR,RAPIDJSON,TBB,TK,VTK,XLIB}
```
Comment (thesamesam) [inline sci-libs/opencascade/opencascade-7.7.1.ebuild:4]: ? And why is this up here?

File: sci-libs/opencascade/opencascade-7.7.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+
+MY_TEST_PV="7.7.0"
+MY_TEST_PV2="${MY_TEST_PV//./_}"
+
+SRC_URI="
+	test? ( https://github.com/Open-Cascade-SAS/${MY_PN}/releases/download/V${MY_TEST_PV2}/${PN}-dataset-${MY_TEST_PV}.tgz
+		-> ${PN}-dataset-${MY_TEST_PV}.tar.gz )
+"
+
+# There's no easy way to test. Testing needs a rather big environment properly set up.
```
Comment (thesamesam) [inline sci-libs/opencascade/opencascade-7.7.1.ebuild:23]: This should be in the standard location, not here.

File: sci-libs/opencascade/opencascade-7.7.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+else
+	MY_PV="${PV//./_}"
+	SRC_URI+="
+		https://github.com/Open-Cascade-SAS/${MY_PN}/archive/refs/tags/V${MY_PV}.tar.gz -> ${P}.tar.gz
+	"
+	S="${WORKDIR}/${MY_PN}-${MY_PV}"
+	KEYWORDS="~amd64 ~arm ~arm64 ~riscv ~x86"
+fi
+
+SLOT="0/$(ver_cut 1-2)"
```
Comment (thesamesam) [inline sci-libs/opencascade/opencascade-7.7.1.ebuild:38]: Swap SLOT/LICENSE.

File: sci-libs/opencascade/opencascade-7.7.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+
+	if use test; then
+		mkdir "${WORKDIR}/data"
+		pushd "${WORKDIR}/data" > /dev/null || die
+		# should be in paths indicated by CSF_TestDataPath environment variable,
+		# or in subfolder data in the script directory
+		unpack "${PN}-dataset-${MY_TEST_PV}.tar.gz"
+		popd > /dev/null || die
+	fi
+
```
Comment (thesamesam) [inline sci-libs/opencascade/opencascade-7.7.1.ebuild:122]: Drop newline

File: sci-libs/opencascade/opencascade-7.7.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+src_install() {
+	cmake_src_install
+
+	doenvd "${T}/99${PN}"
+
+	docompress -x "/usr/share/doc/${PF}/overview/html"
+}
+
+src_test() {
```
Comment (thesamesam) [inline sci-libs/opencascade/opencascade-7.7.1.ebuild:264]: Phase execution order.

**Review by thesamesam — CHANGES_REQUESTED** [review]

File: sci-libs/opencascade/opencascade-7.7.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+			dev-qt/qtwidgets:5
+			dev-qt/qtxml:5
+		)
+	)
+	ffmpeg? ( <media-video/ffmpeg-5:= )
+	freeimage? ( media-libs/freeimage )
+	jemalloc? ( dev-libs/jemalloc )
+	tbb? ( dev-cpp/tbb:= )
+	vtk? (
+		sci-libs/vtk:=[rendering]
```
Comment (vowstar) [inline sci-libs/opencascade/opencascade-7.7.1.ebuild:75]:
```suggestion
		sci-libs/vtk:=[opencascade,rendering]
```
``>=sci-libs/vtk-9.3.0`` require ``opencascade`` use flag

File: sci-libs/opencascade/opencascade-7.8.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,365 @@
...
+			dev-qt/qtwidgets:5
+			dev-qt/qtxml:5
+		)
+	)
+	ffmpeg? ( <media-video/ffmpeg-5:= )
+	freeimage? ( media-libs/freeimage )
+	jemalloc? ( dev-libs/jemalloc )
+	tbb? ( dev-cpp/tbb:= )
+	vtk? (
+		sci-libs/vtk:=[rendering]
```
Comment (vowstar) [inline sci-libs/opencascade/opencascade-7.8.0.ebuild:74]:
```suggestion
		sci-libs/vtk:=[opencascade,rendering]
```
``>=sci-libs/vtk-9.3.0`` require ``opencascade`` use flag

**Pull request closed without merging**